homeassistant/components/simplisafe/__init__.py

```diff
@@ -561,17 +580,20 @@ async def async_init(self) -> None:
             assert self._api.refresh_token
             assert self._api.websocket
 
-        self._api.websocket.add_connect_callback(self._async_websocket_on_connect)
         self._api.websocket.add_event_callback(self._async_websocket_on_event)
-        asyncio.create_task(self._api.websocket.async_connect())
+        self._websocket_reconnect_task = asyncio.create_task(
+            self._async_websocket_connect()
+        )
 
         async def async_websocket_disconnect_listener(_: Event) -> None:
             """Define an event handler to disconnect from the websocket."""
             if TYPE_CHECKING:
                 assert self._api.websocket
 
-            if self._api.websocket.connected:
-                await self._api.websocket.async_disconnect()
+            if self._websocket_reconnect_task:
+                self._websocket_reconnect_task.cancel()
```

Comment:
Should we await the reconnect task after canceling it before we disconnect?

Reply:
Apologies, not sure I understand – is f0678c1 what you're thinking?